backend/config/storage.go

```diff
@@ -0,0 +1,42 @@
+package config
+
+import (
+	"fmt"
+	"os"
+	"strings"
+)
+
+type StorageConfig struct {
+	Path string `yaml:"path"`
+}
+
+func validateStorageConfig(cfg *Config) error {
+	if cfg.Storage == nil {
+		cfg.Storage = &StorageConfig{
+			Path: DefaultStoragePath,
+		}
+	}
+
+	if len(cfg.Storage.Path) == 0 {
+		cfg.Storage.Path = DefaultStoragePath
+	}
+
+	info, err := os.Stat(cfg.Storage.Path)
+	if os.IsNotExist(err) {
+		// if it is a local path, create it
+		if !strings.HasPrefix(cfg.Storage.Path, "/") &&
+		!strings.HasPrefix(cfg.Storage.Path, "../") &&
+		!strings.HasPrefix(cfg.Storage.Path, "~/") {
+			err = os.MkdirAll(cfg.Storage.Path, 0700)
+			if err != nil {
+				return err
+			}
+		} else {
+			return fmt.Errorf("invalid path: \"%s\" does not exist and cannot be created automatically", cfg.Storage.Path)
```

Comment:
It might be better to check permissions using `os.Stat` or `unix.Access` as shown [here](https://stackoverflow.com/a/20026945).

Reply:
In this part what I mean by "cannot be created" is that I don't want to create the directory if it is in some location on the user's FS outside of the confines of this project. Such a directory being specified seems to suggest a mistake and they can create it themself if it isn't a mistake.

I also didn't check to see if the user has write permissions on the directory, but I'll add that.